src/Compilers/CSharp/Test/Emit/PDB/PDBTests.cs

```diff
@@ -1950,6 +1950,138 @@ .maxstack  3
 ", sequencePoints: "C.Main");
         }
 
+        [Fact]
+        [WorkItem(12564, "https://github.com/dotnet/roslyn/issues/12564")]
+        public void ConditionalInAsyncVoidMethod()
+        {
+            var source = @"
+using System;
+
+class Program
+{
+    public static async void Test()
+    {
+        int i = 0;
+
+        if (i != 0)
+            Console
+                .WriteLine();
+    }
+}
+";
+            var v = CompileAndVerify(source, options: TestOptions.DebugDll, additionalRefs: new[]{ MscorlibRef_v4_0_30316_17626, SystemCoreRef, CSharpRef });
+
+            v.VerifyIL("Program.<Test>d__0.System.Runtime.CompilerServices.IAsyncStateMachine.MoveNext()", @"
+{
+  // Code size       81 (0x51)
+  .maxstack  2
+  .locals init (int V_0,
+                bool V_1,
+                System.Exception V_2)
+                           ~IL_0000:  ldarg.0
+                            IL_0001:  ldfld      ""int Program.<Test>d__0.<>1__state""
+                            IL_0006:  stloc.0
+                            .try
+                            {
+                        { ║  -IL_0007:  nop
+               int i = 0; ║  -IL_0008:  ldarg.0
+                              IL_0009:  ldc.i4.0
+                              IL_000a:  stfld      ""int Program.<Test>d__0.<i>5__1""
+              if (i != 0) ║  -IL_000f:  ldarg.0
+                              IL_0010:  ldfld      ""int Program.<Test>d__0.<i>5__1""
+                              IL_0015:  ldc.i4.0
+                              IL_0016:  cgt.un
+                              IL_0018:  stloc.1
+                             ~IL_0019:  ldloc.1
+                              IL_001a:  brfalse.s  IL_0022
+   Console … .WriteLine() ║  -IL_001c:  call       ""void System.Console.WriteLine()""
+                              IL_0021:  nop
+                             ~IL_0022:  leave.s    IL_003c
```

Comment:
I think that would just add more noise. This is clear once one learns the couple of symbols (```-``` and ```~```).